tests/chainerx_tests/unit_tests/routines_tests/test_manipulation.py

```diff
@@ -643,3 +643,90 @@ def forward_xp(self, inputs, xp):
 def test_split_invalid(xp, shape, indices_or_sections, axis):
     a = array_utils.create_dummy_ndarray(xp, shape, 'float32')
     return xp.split(a, indices_or_sections, axis)
+
+
+@op_utils.op_test(['native:0', 'cuda:0'])
+@chainer.testing.parameterize(*(
+    chainer.testing.product({'shapes': [
+        (1,),
+        (1, 1),
+        (1, 1, 1),
+        (2, 2, 2, 2),
+    ],
+        'dtype': chainerx.testing.dtypes.float_dtypes
+    })
+))
+class TestAtleast2d(op_utils.NumpyOpTest):
+
+    dtypes = None
+
+    def setup(self):
+        if numpy.dtype(self.dtype).kind != 'f':
+            self.skip_backward_test = True
+            self.skip_double_backward_test = True
+
+    def generate_inputs(self):
+        a = numpy.random.uniform(0, 1, self.shapes).astype(self.dtype)
+        return a,
+
+    def forward_xp(self, input, xp):
+        x, = input
+        y = xp.atleast_2d(x)
+        return y,
+
+
+@op_utils.op_test(['native:0', 'cuda:0'])
+@chainer.testing.parameterize(*(
+    chainer.testing.product({'shapes': [
+        [(1,)],
+        [(0,), (0,)],
+        [(0, 0,), (0, 0,)],
+        [(1, 0,), (1, 0,)],
+        [(3, 4, 5), (3, 4, 5), (3, 4, 5)],
+        [(2, 3, 2), (2, 3, 2), (2, 3, 2)],
+        [(1, 0, 1), (1, 0, 1), (1, 0, 1)],
+        [(2, 0, 0), (2, 0, 0), (2, 0, 0)],
+        [(1, 0, 1, 0), (1, 0, 1, 0), (1, 0, 1, 0)],
+        [(0, 0, 0, 0), (0, 0, 0, 0), (0, 0, 0, 0)],
+        [(2, 2, 2, 2), (2, 2, 2, 2), (2, 2, 2, 2)],
+    ], 'func': [
+        lambda xp, input: xp.hstack(input),
+        lambda xp, input: xp.vstack(input),
+    ],
+        'dtype': chainerx.testing.dtypes.float_dtypes
+    })
+))
+class TestHVStack(op_utils.NumpyOpTest):
+
+    dtypes = None
+
+    def setup(self):
+        if numpy.dtype(self.dtype).kind != 'f':
+            self.skip_backward_test = True
+            self.skip_double_backward_test = True
+
+    def generate_inputs(self):
+        return _make_inputs(self.shapes, [self.dtype] * len(self.shapes))
+
+    def forward_xp(self, inputs, xp):
+        y = self.func(xp, inputs)
+        return y,
+
+
+@chainerx.testing.numpy_chainerx_array_equal(
+    accept_error=(
+        chainerx.DimensionError, ValueError))
+@pytest.mark.parametrize('shape', [
+    [(2, 1), (1, 2)],
+    [(1, 1, 1), (2, 3, 4)],
+    [(2, 1, 4), (1, 4, 5)],
+    [(1, 1, 2), (3, 5, 8)]
+])
+@pytest.mark.parametrize('func', [
+    lambda xp, input: xp.hstack(input),
+    lambda xp, input: xp.vstack(input),
+])
+def test_hvstack_invalid(func, xp, shape):
+    inputs = _make_inputs(shape, ['float32'] * len(shape))
+    inputs = list(map(xp.array, inputs))
```

Comment:
List comprehension is preferred.
```suggestion
    inputs = [xp.array(a) for a in inputs]
```

Reply:
Ah sure.